### ISSUE TITLE
Configure Customer Pods to Prefer Spot VM Nodes

### ISSUE THREAD

**Issue by axisofentropy** [UffizziCloud/uffizzi_controller#64]
Child of https://github.com/UffizziCloud/uffizzi_platform/issues/899

Add new `affinity` to each Customer Workload `Deployment`. The YAML representation looks like this:
```yaml
apiVersion: apps/v1
kind: Deployment
spec:
  template:
    spec:
      affinity:
        nodeAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - preference:
              matchExpressions:
              - key: cloud.google.com/gke-spot
                operator: In
                values:
                - "true"
            weight: 99
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: sandbox.gke.io/runtime
                operator: In
                values:
                - gvisor
```

The `Affinity` type in our API client is described here: https://pkg.go.dev/k8s.io/api/core/v1#Affinity

Right now we're specifying a `nodeSelector` within each customer `Deployment.spec.template.spec`. From my testing, I think we need to delete this to allow the `affinity` to work correctly.

Test this by looking at YAML representations of `Deployments` created by this controller. Ensure the `affinity` appears exactly as above, and the old `nodeSelector` does not appear.

**Comment by axisofentropy** [comment]
Coordinate release with other children of parent ticket!

**Comment by axisofentropy** [comment]
Implemented by #65
